internal/configfsm/fsm.go

```diff
@@ -0,0 +1,309 @@
+package configfsm
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/qmuntal/stateless"
+)
+
+type Trigger string
+
+const (
+	// Transitions to Missing state
+	StartToMissing           Trigger = "Start->Missing"
+	ClusterNotReadyToMissing Trigger = "ClusterNotReady->Missing"
+	InProgressToMissing      Trigger = "InProgress->Missing"
+	CompletedToMissing       Trigger = "Completed->Missing"
+	TimedOutToMissing        Trigger = "TimedOut->Missing"
+	OutOfDateToMissing       Trigger = "OutOfDate->Missing"
+
+	// Transitions to ClusterNotReady state
+	MissingToClusterNotReady    Trigger = "Missing->ClusterNotReady"
+	InProgressToClusterNotReady Trigger = "InProgress->ClusterNotReady"
+	CompletedToClusterNotReady  Trigger = "Completed->ClusterNotReady"
+	OutOfDateToClusterNotReady  Trigger = "OutOfDate->ClusterNotReady"
+
+	// Transitions to InProgress state
+	ClusterNotReadyToInProgress Trigger = "ClusterNotReady->InProgress"
+	OutOfDateToInProgress       Trigger = "OutOfDate->InProgress"
+	CompletedToInProgress       Trigger = "Completed->InProgress"
```

Comment:
We could have `Missing -> InProgress` if there are no policies at all during the initial config, but we add one as Day2 config, directly in `enforce`.

Reply:
Fixed with the latest updates. Redundant direct states removed. The machine each's state make their own decisions and transition between states until a final state has been reached.